rackspace-server-manager.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+'''
+A script to create and delete the number of servers on Rackspace
+'''
+
+import pyrax
+import time
+import re
+import os
+import argparse
+from Crypto.PublicKey import RSA
+
+def create_key():
+    key = RSA.generate(1024)
+    f = open("private.pem", "wb")
+    f.write(key.exportKey('PEM'))
+    f.close()
+
+    pubkey = key.publickey()
+    f = open("public.pem", "wb")
+    f.write(pubkey.exportKey('OpenSSH'))
+    f.close()
+
+
+def create_node(nova, counts):
+    flavor = nova.flavors.find(name='2 GB General Purpose v1')
+    image = nova.images.find(name='centos7-test')
+    create_key()
+    pubkey = open('public.pem', 'r').read()
+    nova.keypairs.create('distkey', pubkey)
+    for count in range(int(counts)):
+        name = 'distributed-testing.'+str(count+1)
+        node = nova.servers.create(name=name, flavor=flavor.id,
+                                   image=image.id, key_name='distkey')
+
+        while node.status == 'BUILD':
+            time.sleep(5)
+            node = nova.servers.get(node.id)
+
+        ip_address = None
+        for network in node.networks['public']:
+            if re.match('\d+\.\d+\.\d+\.\d+', network):
+                ip_address = network
+                break
+        if ip_address is None:
+            print 'No IP address assigned!'
+            sys.exit(1)
+        print 'The server {0} is waiting at IP address {1}.'.format(count, ip_address)
+
+
+def delete_node(nova, counts):
+    for count in range(int(counts)):
```

Comment:
Use the hosts file to find the machine names rather than guessing based on names. That way, you know exactly which machines to delete.

So you need to add a new parameter to get path to host file name. You can assume a file named hosts in current working directory.